src/core/cgroup.c

```diff
@@ -1397,7 +1410,9 @@ static void cgroup_context_apply(
                         } else
                                 weight = CGROUP_WEIGHT_DEFAULT;
 
-                        cgroup_apply_unified_cpu_weight(u, weight);
+                        cgroup_apply_unified_cpu_idle(u, !weight);
+                        if (weight != CGROUP_WEIGHT_IDLE)
+                                cgroup_apply_unified_cpu_weight(u, weight);
```

Comment:
please move the `weight != CGROUP_WEIGHT_IDLE` check into the `cgroup_apply_unified_cpu_weight()` function and exit early there.